src/pint/models/binary_piecewise.py

```diff
@@ -0,0 +1,465 @@
+"""The BT (Blandford & Teukolsky) model.
+
+See Blandford & Teukolsky 1976, ApJ, 205, 580.
+"""
+import astropy.units as u
+from astropy.table import Table
+from pint.models.parameter import (
+    MJDParameter,
+    floatParameter,
+    prefixParameter,
+    maskParameter,
+)
+import numpy as np
+from pint.toa_select import TOASelect
+from pint import GMsun, Tsun, ls
+from pint.models.parameter import floatParameter
+from pint.models.pulsar_binary import PulsarBinary
+from pint.models.stand_alone_psr_binaries.BT_model import BTmodel
+from pint.models.timing_model import MissingParameter, TimingModel
+from pint.models.stand_alone_psr_binaries.BT_piecewise import BTpiecewise
+
+
+class BinaryBTPiecewise(PulsarBinary):
+    """Model implemenring the BT model.
+
+    This is a PINT pulsar binary BT model class a subclass of PulsarBinary.
+    It is a wrapper for stand alone BTmodel class defined in
+    ./stand_alone_psr_binary/BT_model.py
+    All the detailed calculations are in the stand alone BTmodel.
+    The aim for this class is to connect the stand alone binary model with PINT platform
+    BTmodel special parameters:
+    GAMMA Binary Einsten delay coeeficient
+    """
+
+    register = True
+
+    def __init__(self):
+        super(BinaryBTPiecewise, self).__init__()
+        self.binary_model_name = "BT_piecewise"
+        self.binary_model_class = BTpiecewise
+        self.add_param(
+            floatParameter(
+                name="GAMMA",
+                value=0.0,
+                units="second",
+                description="Time dilation & gravitational redshift",
+            )
+        )
+        self.A1_value_funcs = []
+        self.T0_value_funcs = []
+        self.remove_param("M2")
+        self.remove_param("SINI")
+        self.T0.value = 1
+        self.A1.value = 1
+        self.add_group_range(50000, 51000, frozen=False, j=0)
+        self.add_piecewise_param("T0", 0)
+        self.add_piecewise_param("A1", 0)
+
+    def add_group_range(self, group_start_mjd, group_end_mjd, frozen=True, j=None):
+        """Add an orbital piecewise parameter group range.
+        Parameters
+        ----------
+        group_start_mjd : np.float128
```

Comment:
it would be good to also accept quantities or Times here.  Maybe look at how `DMX` and `SWX` are defined?

Reply:
Changed to accept times and quantities, updated documentation to match